common/src/test/java/com/microsoft/identity/common/MsalCppOAuth2TokenCacheTest.java

```diff
@@ -182,6 +183,97 @@ public void removeAccountTest() throws ClientException {
         Assert.assertNull(restoredAccount);
     }
 
+    @Test
+    public void removeNoMatchingAccount() throws ClientException {
+        // Get the generated account
+        final AccountRecord generatedAccount = mTestBundle.mGeneratedAccount;
+
+        // Save it to the cache
+        mCppCache.saveAccountRecord(generatedAccount);
+        mCppCache.saveCredentials(null, mTestBundle.mGeneratedRefreshToken);
+
+        // Call remove with a different realm
+        final AccountDeletionRecord deletionRecord = mCppCache.removeAccount(
+                generatedAccount.getHomeAccountId(),
+                generatedAccount.getEnvironment(),
+                REALM2
+        );
+
+        // Check the receipt, should remove nothing
+        Assert.assertEquals(0, deletionRecord.size());
+    }
+
+    @Test
+    public void removeAccountWithHomeAccountIdTest() throws ClientException {
+        // Get the generated account
+        final AccountRecord generatedAccount = mTestBundle.mGeneratedAccount;
+
+        // Save it to the cache
+        mCppCache.saveAccountRecord(generatedAccount);
+        mCppCache.saveCredentials(null, mTestBundle.mGeneratedRefreshToken);
+
+        // Call remove
+        final AccountDeletionRecord deletionRecord = mCppCache.removeAccount(
+                generatedAccount.getHomeAccountId(),
+                "",
+                ""
+        );
+
+        // Check the receipt
+        Assert.assertEquals(generatedAccount, deletionRecord.get(0));
+
+        // Try to restore it
+        final AccountRecord restoredAccount = mCppCache.getAccount(
+                generatedAccount.getHomeAccountId(),
+                generatedAccount.getEnvironment(),
+                generatedAccount.getRealm()
+        );
+
+        // Make sure it doesn't exist....
+        Assert.assertNull(restoredAccount);
+    }
+
+    @Test
+    public void removeTwoAccountsWithDifferentRealmsTest() throws ClientException {
+        // Get the generated account
+        final AccountRecord generatedAccount = mTestBundle.mGeneratedAccount;
+
+        // Save it to the cache
+        mCppCache.saveAccountRecord(generatedAccount);
+        mCppCache.saveCredentials(null, mTestBundle.mGeneratedRefreshToken);
+
+        // Save the second account with a different realm
+        generatedAccount.setRealm(REALM2);
+        mCppCache.saveAccountRecord(generatedAccount);
+
+        // Call remove
+        final AccountDeletionRecord deletionRecord = mCppCache.removeAccount(
+                generatedAccount.getHomeAccountId(),
+                "",
+                ""
```

Comment:
I guess we have empty strings because we can't pass `null` for some reason?

Reply:
I don't think there is some special reason. Just because msal cpp is passing it when deleting an account without a realm.